hikari/errors.py

```diff
@@ -227,27 +214,31 @@ class HTTPResponseError(HTTPError):
     raw_body: typing.Any = attr.ib()
     """The response body."""
 
-    reason: typing.Optional[str] = attr.ib(default=None)
-    """The error reason. If `builtins.None`, will generate one automatically."""
-
-    message: str = attr.ib(default=_message_default_factory, init=False)
+    # We want to make it optional here, so ignore mypy complaining
+    message: typing.Optional[str] = attr.ib(default=None)  # type: ignore[assignment]
```

Comment:
This is bad. It breaks typesafety by messing up the contract on the supertype.

We really shouldn't be doing this, cause you'll get an HttpError and expect the message to be a string when you catch it, and youll get None instead.

Reply:
Didn't think about that, will set it to `""` instead of None as a default

Reply:
Why not default to a generic message?

Reply:
This is done in `__str__` anyway, so really no need to complicate it even further and make initializing a error slower :)